L1Trigger/L1THGCal/src/concentrator/HGCalConcentratorSuperTriggerCellImpl.cc

```diff
@@ -5,105 +5,135 @@
 
 HGCalConcentratorSuperTriggerCellImpl::
 HGCalConcentratorSuperTriggerCellImpl(const edm::ParameterSet& conf)
-  : stcSize_(conf.getParameter< std::vector<unsigned> >("stcSize"))
+  : stcSize_(conf.getParameter< std::vector<unsigned> >("stcSize")),
+    fixedDataSizePerHGCROC_(conf.getParameter<bool>("fixedDataSizePerHGCROC"))
 {
 
     if ( stcSize_.size() != kNLayers_ ){
         throw cms::Exception("HGCTriggerParameterError")
             << "Inconsistent size of super trigger cell size vector" << stcSize_.size() ;
     }
     for(auto stc : stcSize_) {
-        if ( stc!=kSTCsizeFine_ && stc!=kSTCsizeCoarse_ ){
-            throw cms::Exception("HGCTriggerParameterError")
-              << "Super Trigger Cell should be of size "<<
-              kSTCsizeFine_ << " or " << kSTCsizeCoarse_;
-        }
-    }
-    
-}
 
-std::map<int,int> 
-HGCalConcentratorSuperTriggerCellImpl::kSplit_ = {
-  {kSTCsizeFine_, kSplit_v8_Fine_},
-  {kSTCsizeCoarse_, kSplit_v8_Coarse_}
-};
+      coarseTCmapping_.checkSizeValidity(stc);
 
-int
-HGCalConcentratorSuperTriggerCellImpl::getSuperTriggerCellId(int detid) const {
+    }
 
+    std::string energyType_(conf.getParameter<string>("type_energy_division"));
```

Comment:
Maybe I got confused about an earlier comment about `energyType_` not needing to be a class member. Or can non-class members also be initialised in the constructor initializer list?

Reply:
Ok, I looked at this line too quickly. I saw the trailing underscore and thought it was a class member.
The trailing underscore should be removed since it is not a class member anymore.